ansible-vault: add page (#3279)
tldr-pages/tldr: pages/common/ansible-vault.md

# ansible-vault

> Encrypts & decrypts values, data structures and files within Ansible projects.
> More information: <https://docs.ansible.com/ansible/latest/user_guide/vault.html#id17>.

> - Create a new encrypted vault file with a prompt for a password:

Reviewer: Perhaps "Create a new encrypted vault file; user will be prompted for a password:" sonuds better ? "with a prompt" reads a bit weird to me. And this also makes it same as the `psql` page.

> - Create a new encrypted vault file with vault prompt for password:

Reviewer 2: Perhaps this will read better?
```suggestion
- Create a new encrypted vault file with a prompt for a password:
```

> `ansible-vault create {{vault_file}}`

Reviewer 2: Ideally this would be `path/to/vault_file`, but I think we want to keep it consistent across the entire page first - and with other pages second. In this case, we should probably leave it as-is because if we were to change the later examples, they would get rather long.

> - Create a new encrypted vault file using vault key file to encrypt it:

Reviewer 2: Perhaps this will read better?
```suggestion
- Create a new encrypted vault file using a vault key file to encrypt it:
```

> - Create a new encrypted vault file with vault key file to encrypt with:

Reviewer 2: Perhaps this will read better?
```suggestion
- Create a new encrypted vault file using vault key file to encrypt it:
```

`ansible-vault create --vault-password-file={{password_file}} {{vault_file}}`

> - Encrypt existing file, using password file (optional):

Reviewer 2: How does this example differ from the previous one? Both of the commands look the same to me.

Reviewer 3: It encrypts an existing file, unlike the other one which creates a new file aswell.

Reviewer 2: Ah, I see. What about this then?
```suggestion
- Encrypt an existing file using an optional password file:
```

`ansible-vault encrypt --vault-password-file={{password_file}} {{vault_file}}`

> - Encrypt a string using ansible's encrypted string format using prompts:

Reviewer 2: Perhaps this will read better?
```suggestion
- Encrypt a string using Ansible's encrypted string format, displaying interactive prompts:
```

`ansible-vault encrypt_string`

> - View an encrypted file, using password file to decrypt:

Reviewer 2: Perhaps this will read better?
```suggestion
- View an encrypted file, using a password file to decrypt:
```

`ansible-vault view --vault-password-file={{password_file}} {{vault_file}}`

> - Rekey already encrypted vault file with new password file:

Reviewer 2: Perhaps this will read better?
```suggestion
- Re-key already encrypted vault file with a new password file:
```
